src/librustc_mir/interpret/eval_context.rs

```diff
@@ -106,9 +105,17 @@ pub enum StackPopCleanup {
     None { cleanup: bool },
 }
 
-// State of a local variable
+/// State of a local variable including a memoized layout
+#[derive(Clone, PartialEq, Eq)]
+pub struct LocalValue<'tcx, Tag=(), Id=AllocId> {
```

Comment:
Nit: I think swapping the names `LocalValue` and `LocalState` is nicer. The `layout` stuff looks much more state-y than value-y to me.